app/src/main/java/org/wikipedia/games/onthisday/OnThisDayGameBaseFragment.kt

```diff
@@ -58,90 +56,66 @@ abstract class OnThisDayGameBaseFragment : Fragment() {
     protected fun prepareAndOpenArchiveCalendar(viewModel: OnThisDayGameViewModel) {
         lifecycleScope.launch {
             val startDateBasedOnLanguage = LANG_CODES_SUPPORTED.associateWith { dateReleasedForLang(it) }
-            val localDate = startDateBasedOnLanguage[viewModel.wikiSite.languageCode]
-            val startDate = Date.from(localDate?.atStartOfDay(ZoneId.systemDefault())?.toInstant())
+            val startDate = startDateBasedOnLanguage[viewModel.wikiSite.languageCode]!!
```

Comment:
Instead of having a `!!` for the startDate, would it be possible to handle this properly?